.gitlab/binary_build/otel_agent.yml

```diff
@@ -0,0 +1,25 @@
+---
+.build_otel_agent_binary_common:
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/deb_x64$DATADOG_AGENT_BUILDIMAGES_SUFFIX:$DATADOG_AGENT_BUILDIMAGES
+  stage: binary_build
+  rules:
+    - !reference [.except_mergequeue]
+    - when: on_success
+  script:
+    - !reference [.retrieve_linux_go_deps]
+    - inv -e otel-agent.build
+  needs: ["go_deps"]
```

Comment:
Should we add lint and go mod tidy checks ?

Reply:
We certainly can, I don't have much of an opinion about that. I've looked at other binary build jobs and they're a bit inconsistent in this regard. Note that the only effect of adding them here is that this job will wait until those other jobs finish successfully, which is not strictly necessary (whereas `go_deps` are definitely necessary I think).